mingw: build error due to function redifinitions

> src/cpp/icompare.hpp:8: error: redefinition of 'int strcasecmp(const char*, const char*)'
> src/cpp/icompare.hpp:13: error: redefinition of 'int strncasecmp(const char*, const char*, size_t)'

https://github.com/thelink2012/gta3sc/blob/master/src/cpp/icompare.hpp#L8
https://github.com/thelink2012/gta3sc/blob/master/src/cpp/icompare.hpp#L13

The problem is the `#if defined(_WIN32)` guard getting triggered by mingw even though mingw does implement strcasecmp and strncasecmp

--- a/src/cpp/icompare.hpp
+++ b/src/cpp/icompare.hpp
@@ -4,7 +4,7 @@
 #include <string>
 #include <cstring>
 
-#if defined(_WIN32)
+#if defined(_MSC_VER)
 inline int strcasecmp(const char* a, const char* b)
 {
     return _stricmp(a, b);
@@ -14,7 +14,7 @@ inline int strncasecmp(const char* a, const char* b, size_t count)
 {
     return _strnicmp(a, b, count);
 }
-#elif defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
+#elif defined (__unix__) || (__MINGW32__) || (defined (__APPLE__) && defined (__MACH__))
     // Already has strcasecmp and strncasecmp
 #else
 #   error

